test/compress/issue-203.js

```diff
@@ -26,9 +26,7 @@ compress_new_function_with_destruct: {
     }
     expect: {
         Function("a", "[b]", "return a");
-        Function("a", "{bb}", "return a");
-        Function("[[a]]", "[{bb}]", 'return a');
+        Function("a", "{bb:b}", "return a");
+        Function("[[a]]", "[{bb:b}]", 'return a');
```

Comment:
I think the more fundamental problem is this optimisation is making a wrong assumption when it transform `new Function("foo, {bar}", "return foo + bar;");` into:
```js
(function(foo, {bar}) {
    return foo + bar;
})()
```
Because `uglify-js` is so advanced now, we literally will substitute `foo` and `{bar}` and then evaluated `foo + bar` to a constant, which is why `side_effects` then proceed to erase the whole function.